### ISSUE TITLE
The argument type 'HttpClientResponse' can't be assigned to the parameter type 'Stream<Uint8List>'

### ISSUE THREAD

**Issue by pushangupta** [flutterchina/dio#360]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by pushangupta** [comment]
`file:///Users/mac/flutter/.pub-cache/hosted/pub.dartlang.org/dio-2.1.9/lib/src/adapter.dart:161:7: Error: The argument type 'HttpClientResponse' can't be assigned to the parameter type 'Stream<Uint8List>'.
`

I just upgraded flutter and this error showed

**Comment by renjun2012** [comment]
me too.

/.pub-cache/hosted/pub.flutter-io.cn/dio-2.1.9/lib/src/adapter.dart:161:7: Error: The argument type 'HttpClientResponse' can't be assigned to the parameter type 'Stream<Uint8List>'.
 - 'HttpClientResponse' is from 'dart:_http'.
 - 'Stream' is from 'dart:async'.
 - 'Uint8List' is from 'dart:typed_data'.
Try changing the type of the parameter, or casting the argument to 'Stream<Uint8List>'.
      responseStream,
      ^

**Comment by pushangupta** [comment]
This has broken my entire app :(

**Comment by synw** [comment]
Same here with a Dart package, not a Flutter app

**Comment by keyduq** [comment]
I have the same issue

**Comment by pushangupta** [comment]
I managed to build the app by casting 

`responseStream as Stream<Uint8List>,`

on line 161.

But now I get this error:

`[ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: DioError [DioErrorType.DEFAULT]: RangeError (index): Invalid value: Only valid value is 0: 1#0      List.[] (dart:core-patch/growable_array.dart:145:60)`

**Comment by keyduq** [comment]
> 
> 
> I managed to build the app by casting
> 
> `responseStream as Stream<Uint8List>,`
> 
> on line 161.
> 
> But now I get this error:
> 
> `[ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: DioError [DioErrorType.DEFAULT]: RangeError (index): Invalid value: Only valid value is 0: 1#0 List.[] (dart:core-patch/growable_array.dart:145:60)`

Casting it to `Stream<List<int>>` according to this breaking change https://github.com/dart-lang/co19/issues/383 works on my side

**Comment by pushangupta** [comment]
@keyduq Could you describe a little?

**Comment by keyduq** [comment]
From what I understand the issue is because there is a breaking change on Dart language where `HttpRequest` and `HttpClientResponse` now returns `Stream<UInt8List>` instead of `Stream<List<int>>`, the workaround is to cast `Stream<UInt8List>` as `Stream<List<int>>` so I just did the same thing that you do:

**Comment by pushangupta** [comment]
@keyduq It still doesn't make any requests. I made change on line 161 in adapter file. Did you do the same just instead of Stream<UInt8List>, you did Stream<List<int>>?

**Comment by keyduq** [comment]
@pushangupta I had change the line 161 in adapter.dart to
`responseStream as Stream<List<int>>,`

**Comment by pushangupta** [comment]
@keyduq strange! It doesn't work for me

**Comment by badrobot15** [comment]
I was moments away from launching my app on the Play Store when I got this error on build. My error was thrown in adapter.dart. I followed the commit and undid the changes made to Stream. I came here to unload my frustrations when I saw @keyduq solution. I can confirm undoing the commit in adapter.dart will solve the problem. To the one who made the commit, were you the one at cloudflare the other day?

**Comment by keyduq** [comment]
@pushangupta try this steps:

1. Make sure that you set Dio version to explicitly 2.1.8 in `pubspec.yaml` set `dio: 2.1.8`
2. Change the line 161 of adapter.dart in `flutter/.pub-cache/hosted/pub.dartlang.org/dio-2.1.8/lib/src` to `responseStream as Stream<List<int>>,`

With version 2.1.9 seems to not work out this workaround.

**Comment by svmvnthvs** [comment]
@keyduq  Hi, I tried your solution and it solved this issue but afterwards I get this error.

`Unhandled Exception: DioError [DioErrorType.DEFAULT]: type '_HttpClientResponse' is not a subtype of type 'Stream<Uint8List>'#0      DefaultHttpClientAdapter.fetch (package:dio/src/adapter.dart:161:22)`

**Comment by pushangupta** [comment]
@keyduq Yes thats the difference between yours and mine :)

**Comment by JunevaYang** [comment]
urgency

**Comment by hello-acuario** [comment]
@keyduq I tried your solution on version 2.1.9 and it works!!

**Comment by williamjohn2008** [comment]
I have changed the line 161 in adapter.dart to responseStream as Stream<List<int>>,
then  DioError [DioErrorType.DEFAULT]: type '_HttpClientResponse' is not a subtype of type 'Stream<Uint8List>'#0

**Comment by miguelpruivo** [comment]
This should be flagged as a _**breaking change**_ in the changelog for the version `2.1.9`.

A quick workaround as of now, if you don't want to refactor your code, is to set your dio version to `2.1.8` instead of `^2.x.x`.

**Comment by williamjohn2008** [comment]
> refactor

thank you

**Comment by winter91** [comment]
me too

**Comment by amake** [comment]
> This should be flagged as a _**breaking change**_ in the changelog for the version `2.1.9`.

Merely flagging in the changelog wouldn't prevent breakage; I believe it should have been e.g. `2.2.0`, not `2.1.9` to begin with.

> A quick workaround as of now, if you don't want to refactor your code, is to set your dio version to `2.1.8` instead of `^2.x.x`.

`dio` is used by [`catcher`](https://pub.dev/packages/catcher), which is how this issue hit me.

If you are not directly using `dio` and don't want to add it as a direct dependency, you can override its version as a transitive dependency in `pubspec.yaml` like so:

```yaml
dependency_overrides:
  dio: 2.1.8
```

**Comment by thangtq-intelads** [comment]
me too :(, pls fix it

**Comment by Cvilia** [comment]
I tried to change dio to 2.1.8,and modified line 161 in adapter.dart to responseStream as Stream<List>and Stream<List<int>>,my project could build ,but when I initiated net request through dio,then it reminded me of an unknown error.
I didn't know but guessed this unknown error was related to this bug.
Hope it can be fixed soon.

**Comment by sm9i** [comment]
DioError [DioErrorType.DEFAULT]: type '_StreamHandlerTransformer<List<int>, List<int>>' is not a subtype of type 'StreamTransformer<Uint8List, List<int>>' of 'streamTransformer'#0   
]我也是

**Comment by allencelee** [comment]
> @pushangupta try this steps:
> 
> 1. Make sure that you set Dio version explicitly to 2.1.8, in `pubspec.yaml` set `dio: 2.1.8`
> 2. Change the line 161 of adapter.dart in `flutter/.pub-cache/hosted/pub.dartlang.org/dio-2.1.8/lib/src` to `responseStream as Stream<List<int>>,`
> 
> With version 2.1.9 seems to not work out this solution.



> @pushangupta try this steps:
> 
> 1. Make sure that you set Dio version explicitly to 2.1.8, in `pubspec.yaml` set `dio: 2.1.8`
> 2. Change the line 161 of adapter.dart in `flutter/.pub-cache/hosted/pub.dartlang.org/dio-2.1.8/lib/src` to `responseStream as Stream<List<int>>,`
> 
> With version 2.1.9 seems to not work out this solution.
Thank you for solving my problem

**Comment by Cvilia** [comment]
> @pushangupta I have changed the line 161 in adapter.dart to
> `responseStream as Stream<List<int>>,`
> and works on my side.

Does dio could initiate network requests?
I try it in my project ,but it tip me unknown error

**Comment by wendux** [comment]
I'm sorry for this.  This is because the latest version of flutter's SDK has changed the type of HttpClientResponse https://github.com/dart-lang/co19/issues/383, and I have committed a patch for this in dio 2.1.10 .  the change is here https://github.com/flutterchina/dio/blob/a2ade28e1063716d4ee0b2fa6aca723ea2ab9221/package_src/lib/src/adapter.dart#L170

**Comment by pushangupta** [comment]
@wendux Build 2.1.10 seems to be on commit `"c7880ae"`. That build is still failing in my app. However you latest commit `"a2ade28"` is working perfectly fine. Please issue another build version as `2.1.11 ` or you could also change the middle order to `2.2.0` as this fixes the major breaking change.
